src/chart/SunburstChart.tsx

```diff
@@ -0,0 +1,188 @@
+  endAngle?: number;
+  children?: React.ReactNode;
+  fill?: string;
+  stroke?: string;
+  /* an object with svg text options to control the appearance of the chart labels. */
+  textOptions?: TextOptions;
+
+  onMouseEnter?: (node: SunburstData, e: any) => void;
+
+  onMouseLeave?: (node: SunburstData, e: any) => void;
+
+  onClick?: (node: SunburstData) => void;
+}
+
+interface DrawArcOptions {
+  radius: number;
+  innerR: number;
+  initialAngle: number;
+  childColor?: string;
+}
+
+const defaultTextProps = {
+  fontFamily: 'sans-serif',
+  fontWeight: 'bold',
+  paintOrder: 'stroke fill',
+  fontSize: '.75rem',
+  stroke: '#FFF',
+  fill: 'black',
+};
+
+function getMaxDepthOf(node: SunburstData): number {
+  if (!node.children || node.children.length === 0) return 1;
+
+  // Calculate depth for each child and find the maximum
+  const childDepths = node.children.map(d => getMaxDepthOf(d));
+  return 1 + Math.max(...childDepths);
+}
+
+export const SunburstChart = ({
+  className,
+  data,
+  children,
+  width,
+  height,
+  padding = 2,
+  ringPadding = 2,
+  innerRadius = 50,
+  fill = '#333',
+  stroke = '#FFF',
+  textOptions = defaultTextProps,
+  outerRadius = Math.min(width, height) / 2,
+  cx = width / 2,
+  cy = height / 2,
+  startAngle = 0,
+  endAngle = 360,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+}: SunburstChartProps) => {
+  const [isTooltipActive, setIsTooltipActive] = useState(false);
+  const [activeNode, setActiveNode] = useState<SunburstData | null>(null);
+
+  const rScale = scaleLinear([0, data.value], [0, endAngle]);
+  const treeDepth = getMaxDepthOf(data);
+  const thickness = (outerRadius - innerRadius) / treeDepth;
+
+  const sectors: React.ReactNode[] = [];
+
+  // event handlers
+  function handleMouseEnter(node: SunburstData, e: any) {
+    if (onMouseEnter) onMouseEnter(node, e);
+    setActiveNode(activeNode);
+    setIsTooltipActive(true);
+  }
+
+  function handleMouseLeave(node: SunburstData, e: any) {
+    if (onMouseLeave) onMouseLeave(node, e);
+    setActiveNode(null);
+    setIsTooltipActive(false);
+  }
+
+  function handleClick(node: SunburstData) {
+    if (onClick) onClick(node);
+  }
+
+  // recursively add nodes for each data point and its children
+  function drawArcs(childNodes: SunburstData[] | undefined, options: DrawArcOptions): any {
+    const { radius, innerR, initialAngle, childColor } = options;
+
+    let currentAngle = initialAngle;
+
+    if (!childNodes) return; // base case: no children of this node
+
+    childNodes.forEach(d => {
+      const arcLength = rScale(d.value);
+      const start = currentAngle;
+      // color priority - if there's a color on the individual point use that, otherwise use parent color or default
+      const fillColor = d?.fill ?? childColor ?? fill;
+      const { x: textX, y: textY } = polarToCartesian(0, 0, innerR + radius / 2, -(start + arcLength - arcLength / 2));
+      currentAngle += arcLength;
+      sectors.push(
+        <g>
```

Comment:
wonder if there are any accessibility concerns we can take care of up front? aria-labels, tabIndex, etc.?

Reply:
that's a good idea - I set these attributes on the g element that wraps each data point, and tested that the chart is now keyboard accessible